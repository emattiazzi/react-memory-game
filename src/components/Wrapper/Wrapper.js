import React from 'react';
import PropTypes from 'prop-types';

Wrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default function Wrapper({ children }) {
  return (
    <div
      style={{
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'center',
        maxWidth: '860px',
        margin: '0 auto',
      }}
    >
      {children}
    </div>
  );
}
