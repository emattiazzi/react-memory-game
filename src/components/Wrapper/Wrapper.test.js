import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Wrapper from './Wrapper';

configure({ adapter: new Adapter() });

const Noop = () => <div></div>

describe('<Wrapper />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Wrapper onClick={() => {}}>String</Wrapper>);
    expect(toJson(tree)).toMatchSnapshot();
  })
  it('renders a children string node', () => {
    const wrapper = shallow(<Wrapper onClick={() => {}}>Text</Wrapper>);
    expect(typeof wrapper.props().children).toBe('string');
    expect(wrapper.props().children).toBe('Text');
  })
  it('renders a children node', () => {
    const wrapper = shallow(<Wrapper onClick={() => {}}><Noop /></Wrapper>);
    expect(typeof wrapper.props().children).toBe('object');
    expect(wrapper.props().children).toEqual(<Noop />);
  })
})