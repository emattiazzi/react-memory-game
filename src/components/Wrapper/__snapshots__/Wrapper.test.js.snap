// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> matches the snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "margin": "0 auto",
      "maxWidth": "860px",
    }
  }
>
  String
</div>
`;
